sys/boot/geli/geliboot.c

```diff
@@ -27,53 +27,75 @@
  * $FreeBSD$
  */
 
-#include <crypto/intake.h>
-
+#include "geliboot_internal.h"
 #include "geliboot.h"
 
 SLIST_HEAD(geli_list, geli_entry) geli_head = SLIST_HEAD_INITIALIZER(geli_head);
 struct geli_list *geli_headp;
 
-typedef u_char geli_mkey[G_ELI_DATAIVKEYLEN];
+typedef u_char geli_ukey[G_ELI_USERKEYLEN];
 
-static geli_mkey saved_keys[GELI_MAX_KEYS];
+static geli_ukey saved_keys[GELI_MAX_KEYS];
 static unsigned int nsaved_keys = 0;
 
+/*
+ * Copy keys from local storage to the keybuf struct.
+ * Destroy the local storage when finished.
+ */
 void
-geli_fill_keybuf(struct keybuf *keybuf)
+geli_fill_keybuf(struct keybuf *fkeybuf)
```

Comment:
renamed to avoid shadowing the struct name, silences a warning from the compiler